Project version manipulation not working (?!?)

I have set the _pom-manipulation-ext_ into _.mvn\extensions.xml_ with 
```
<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
  <extension>
    <groupId>org.commonjava.maven.ext</groupId>
    <artifactId>pom-manipulation-ext</artifactId>
    <version>4.3</version>
  </extension>
</extensions>
```

From the root of the multi-project project,  have then run the following:
`mvn -DversionSuffix=Some-Suffix clean install -e -B -f pom.xml -pl :project-common -am -P<some profiles> -s settings-maven.xml`

In the target folder, the package remains: _project-common-3.4.0-SNAPSHOT.jar_ and not _project-common-3.4.0-SNAPSHOT-Some-Suffix.jar_

Should not it behave as if having the _pom-manipulation-ext-4.3.jar_ was in the _%MAVEN_HOME\lib\ext_ directory?

One step forward, two back. I can now get it to pick up the extension within `.mvn/extensions.xml` but it gets classloader issues e.g.
```mvn -DversionSuffix=redhat-1
[INFO] Running Maven Manipulation Extension (PME) 4.4-SNAPSHOT ( SHA: 98d8b624 )
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.InitialGroovyManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RangeResolver
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RESTBOMCollector
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.ProfileInjectionManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.SuffixManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RelocationManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RESTCollector
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.ProjectVersioningManipulator
[INFO] Version Manipulator: Calculating the necessary versioning changes.
[INFO] Looking for new version: org.commonjava.maven.ext.integration-test:simple-numeric-override:1.0 (found: 1.0.0.redhat-1)
[INFO] Changed main version in org.commonjava.maven.ext.integration-test:simple-numeric-override:1.0.0.redhat-1 [pom=/home/rnc/Work/PME/pom-manipulation-ext/integration-test/target/it-ext/simple-numeric-with-maven-extensions/pom.xml]
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.ParentInjectionManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.PropertyManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.PluginManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.DependencyManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RepoAndReportingRemovalManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.DependencyRemovalManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.PluginRemovalManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.NexusStagingMavenPluginRemovalManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.ProfileRemovalManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.PluginInjectingManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.RepositoryInjectionManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.ProjectVersionEnforcingManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.DistributionEnforcingManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.BOMBuilderManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.JSONManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.XMLManipulator
[INFO] Running manipulator org.commonjava.maven.ext.core.impl.FinalGroovyManipulator
[INFO] Maven-Manipulation-Extension: Rewrite changed: [org.commonjava.maven.ext.integration-test:simple-numeric-override:1.0.0.redhat-1 [pom=/home/rnc/Work/PME/pom-manipulation-ext/integration-test/target/it-ext/simple-numeric-with-maven-extensions/pom.xml]]
### PClassloader : ClassRealm[plexus.core, parent: null] and class org.codehaus.plexus.util.xml.Xpp3Dom and 238308051
### XClassloader : ClassRealm[coreExtension>org.commonjava.maven.ext:pom-manipulation-ext:4.4-SNAPSHOT, parent: ClassRealm[plexus.core, parent: null]] and class org.codehaus.plexus.util.xml.Xpp3Dom and 1665525689
[ERROR] Extension failure
java.lang.ClassCastException: org.codehaus.plexus.util.xml.Xpp3Dom cannot be cast to org.codehaus.plexus.util.xml.Xpp3Dom
    at org.commonjava.maven.ext.common.jdom.JDOMModelConverter.updatePlugin (JDOMModelConverter.java:1511)
    at org.commonjava.maven.ext.common.jdom.JDOMModelConverter.iteratePlugin (JDOMModelConverter.java:537)
```

(cc @dwalluck )

--- a/README.md
+++ b/README.md
@@ -15,7 +15,7 @@
 
 POM Manipulation Extension (PME) is a Maven tool to align the versions in your POMs according to some external reference, sort of like a BOM but much more extensive and without the added baggage of a BOM declaration.
 
-It is supplied as a core library, a Maven extension (in the sense of installing to `lib/ext`, not [pom extensions](https://maven.apache.org/pom.html#Extensions)) and a command line tool.
+It is supplied as a core library, a Maven extension (either by installing to `lib/ext`, or using as a [pom extensions](https://maven.apache.org/pom.html#Extensions)) and a command line tool.
 
 This extension combines many of the features of [VMan](https://github.com/jdcasey/pom-version-manipulator), [Maven Versioning Extension](https://github.com/jdcasey/maven-versioning-extension) and [Maven Dependency Management Extension](https://github.com/jboss/maven-dependency-management-extension).
 

--- a/ext/pom.xml
+++ b/ext/pom.xml
@@ -33,6 +33,12 @@
     <dependency>
       <groupId>org.commonjava.maven.ext</groupId>
       <artifactId>pom-manipulation-common</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.commonjava.maven.ext</groupId>
@@ -44,6 +50,11 @@
     </dependency>
 
     <!-- External dependencies -->
+    <dependency>
+      <groupId>org.codehaus.plexus</groupId>
+      <artifactId>plexus-utils</artifactId>
+      <scope>provided</scope>
+    </dependency>
 
     <dependency>
       <groupId>javax.inject</groupId>
@@ -62,26 +73,62 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-artifact</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-compat</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-model</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-model-builder</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-settings</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.eclipse.aether</groupId>
@@ -91,6 +138,12 @@
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-utils</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>

--- a/integration-test/pom.xml
+++ b/integration-test/pom.xml
@@ -333,6 +333,28 @@
                   <settingsFile>target/it-${testMode}/settings.xml</settingsFile>
                 </configuration>
               </execution>
+              <!-- Need to run as separate phase as we don't want the built in Maven with custom extension -->
+              <execution>
+                <id>integration-test-pom-extension</id>
+                <goals>
+                  <goal>integration-test</goal>
+                  <goal>verify</goal>
+                </goals>
+                <configuration>
+                  <debug>true</debug>
+                  <goals>
+                    <goal>package</goal>
+                  </goals>
+                  <mavenHome>${maven.home}</mavenHome>
+                  <pomIncludes>
+                    <pomInclude>pom-extension/pom.xml</pomInclude>
+                  </pomIncludes>
+                  <projectsDirectory>${project.build.directory}/it-${extIter}</projectsDirectory>
+                  <preBuildHookScript>setup</preBuildHookScript>
+                  <postBuildHookScript>verify</postBuildHookScript>
+                  <settingsFile>target/it-${testMode}/settings.xml</settingsFile>
+                </configuration>
+              </execution>
             </executions>
           </plugin>
           <plugin>

--- a/integration-test/src/it/pom-extension/.mvn/extensions.xml
+++ b/integration-test/src/it/pom-extension/.mvn/extensions.xml
@@ -1,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2012 Red Hat, Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<extensions>
+    <extension>
+        <groupId>org.commonjava.maven.ext</groupId>
+        <artifactId>pom-manipulation-ext</artifactId>
+        <version>${project.version}</version>
+    </extension>
+</extensions>

--- a/integration-test/src/it/pom-extension/invoker.properties
+++ b/integration-test/src/it/pom-extension/invoker.properties
@@ -1,0 +1,17 @@
+#
+# Copyright (C) 2012 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+invoker.goals=package

--- a/integration-test/src/it/pom-extension/pom.xml
+++ b/integration-test/src/it/pom-extension/pom.xml
@@ -1,0 +1,57 @@
+<!--
+
+    Copyright (C) 2012 Red Hat, Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.commonjava.maven.ext.integration-test</groupId>
+  <artifactId>pom-extension</artifactId>
+  <version>1.0.jbossorg-101909</version>
+  <packaging>jar</packaging>
+
+  <name>simple</name>
+  <url>http://maven.apache.org</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
+          <configuration>
+            <source>1.8</source>
+            <target>1.8</target>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+</project>

--- a/integration-test/src/it/pom-extension/test.properties
+++ b/integration-test/src/it/pom-extension/test.properties
@@ -1,0 +1,18 @@
+#
+# Copyright (C) 2012 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+versionSuffix=redhat-1
+versionSuffixStrip=

--- a/integration-test/src/it/pom-extension/verify.groovy
+++ b/integration-test/src/it/pom-extension/verify.groovy
@@ -1,0 +1,22 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+def pomFile = new File( basedir, 'pom.xml' )
+System.out.println( "Slurping POM: ${pomFile.getAbsolutePath()}" )
+
+def pom = new XmlSlurper().parse( pomFile )
+System.out.println( "POM Version: ${pom.version.text()}" )
+
+assert pom.version.text().equals( '1.0.0.redhat-1' )

--- a/integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/ITestUtils.java
+++ b/integration-test/src/main/java/org/commonjava/maven/ext/integrationtest/ITestUtils.java
@@ -89,6 +89,7 @@ public class ITestUtils
             }
         }
         add( "groovy-manipulator-first-http" );
+        add( "pom-extension" );
     }};
 
     static final Map<String, String> LOCATION_REWRITE = new HashMap<String, String>()

